tests/tools/test_mlflow_query.py

```
@@ -0,0 +1,494 @@
+    def test_extra_params_listed_under_fields_section(self):
+        run = _make_run(params={"actual output": "ans", "failure": "none"})
+        result = format_fields([run], "my-run")
+        assert "## Params (pass to --fields)" in result
+        assert "- actual output" in result
+        assert "- failure" in result
+
+    def test_auto_params_excluded_from_fields_section(self):
+        run = _make_run(params={"question": "Q?", "question_id": "1", "failure": "x"})
+        result = format_fields([run], "my-run")
+        lines = result.splitlines()
+        fields_start = next(i for i, l in enumerate(lines) if "pass to --fields" in l)
```

Ruff (0.15.10)

[error] Ambiguous variable name: `l`

(E741)